ddl/ddl_api.go

```diff
@@ -1938,6 +1965,14 @@ func setDefaultValue(ctx sessionctx.Context, col *table.Column, option *ast.Colu
 	if err != nil {
 		return ErrColumnBadNull.GenWithStack("invalid default value - %s", err)
 	}
+	if _, value, err = checkColumnDefaultValue(ctx, col, value); err != nil {
```

Comment:
Why add this?

Reply:
alter column should also check the default value.